### ISSUE TITLE
Disable weight decay for certain layers

### ISSUE THREAD

**Issue by ndryden** [LLNL/lbann#809]
We should be able to selectively disable the application of weight decay to certain layers/weights. For example, ResNet training often does not apply weight decay to batch normalization parameters.
